src/System.IO.Pipelines/PipelineReader.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    /// <summary>
+    /// Represents a pipeline from which data can be read.
+    /// </summary>
+    public abstract class PipelineReader : IPipelineReader
+    {
+        /// <summary>
+        /// The underlying <see cref="PipelineReaderWriter"/> the <see cref="PipelineReader"/> communicates over.
+        /// </summary>
+        protected readonly PipelineReaderWriter _input;
```

Comment:
Yeah, the fact that reader wraps reader/writers shows that we don't yet have a good name for the reader/writer.